feat(@aws-amplify/geo): Add docs for new Geo category in javascript (#3435)
aws-amplify/docs: docs/lib/geo/fragments/js/maps.md

> Prerequisite: [Install and configure](~/cli/start/install.md) the Amplify CLI

## Provision a map resource

The primary way to provision Geo map resources is through the Amplify CLI.

Note: You will need the `@geo` release of the CLI for this. For more information, please see [Getting started](~/lib/geo/getting-started.md)

```bash
amplify add geo
```

```
? Select which capability you want to add: (Use arrow keys)
❯ Map (visualize the geospatial data)
  Location search (search by places, addresses, coordinates)
```

From here you can follow the prompts to generate your new Map resource. Make sure to push your changes up after configuration:

```
amplify push
```

For more information, you can visit the full [Amplify CLI Geo Maps docs](~/cli/geo/maps.md).

> 

Reviewer: All this can be summarized in one sentence and moved below Display a Map.:
"First, make sure you've provisioned a map resource by running `amplify add geo`, selecting the **Map (visualize the geospatial data)** option and running `amplify push` to depoly."

Author: Updated in both maps and search

## Display a map

> Add maplibre-gl-js to your app with `yarn` or `npm`:
> 
> ```bash
> npm install -S maplibre-gl
> ```
> [MapLibre GL](https://github.com/maplibre/maplibre-gl-js) is an open source map rendering library for javascript.
> 

Reviewer: ```suggestion
To render a map, the [MapLibre GL](https://github.com/maplibre/maplibre-gl-js) library is required. MapLibre GL is an open source map rendering library for JavaScript. 

Add the `maplibre-gl-js` package to your app:

```bash
npm install -S maplibre-gl
```

```

Add maplibre-gl-js-amplify to your app with `yarn` or `npm`:

```bash
npm install -S maplibre-gl-js-amplify
```

> [maplibre-gl-js-amplify](https://github.com/aws-amplify/maplibre-gl-js-amplify) is a library that makes it easy to integration MapLibre with Amplify Geo.

Reviewer: Remove.

Author: removed

> In your app’s entry point i.e. `App.js`, import and load the configuration file `aws-exports.js` which has been created and replaced into `/src` folder in the previous step.

Reviewer: Next, configure the Amplify library in your app. All geo-related configuration information is available within the `aws-exports.js` file. Add the following to your app's entry point (i.e. `App.js`):

Author: Updated.

```javascript
import Amplify from 'aws-amplify';
import { AmplifyMapLibreRequest } from "maplibre-gl-js-amplify";
import { Map } from "maplibre-gl";
import awsconfig from './aws-exports';

Amplify.configure(awsconfig);
```

> In your app create an async function to call [AmplifyMapLibreRequest's](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#amplifymaplibrerequest) function `createMapLibreMap` to create a new instance of maplibre-gl [Map](https://maplibre.org/maplibre-gl-js-docs/api/map/).

Reviewer: ```suggestion
Next, render the [Map](https://maplibre.org/maplibre-gl-js-docs/api/map/) with the help of [AmplifyMapLibreRequest's](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#amplifymaplibrerequest).
```

In your app’s entry point i.e. `App.js`, import and load the configuration file `aws-exports.js` which has been created and replaced into `/src` folder in the previous step.
```javascript
import Amplify from 'aws-amplify';
import { AmplifyMapLibreRequest } from "maplibre-gl-js-amplify";
import { Map } from "maplibre-gl";
import awsconfig from './aws-exports';

Amplify.configure(awsconfig);
```

In your app create an async function to call [AmplifyMapLibreRequest's](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#amplifymaplibrerequest) function `createMapLibreMap` to create a new instance of maplibre-gl [Map](https://maplibre.org/maplibre-gl-js-docs/api/map/).
```javascript
async function initializeMap() {
    const map = await AmplifyMapLibreRequest.createMapLibreMap({
        container: "map", // An HTML Element or HTML element ID to render the map in https://maplibre.org/maplibre-gl-js-docs/api/map/
        center: [-123.1187, 49.2819],
        zoom: 11,
        region: "us-west-2"
    })
}

initializeMap();
```

![A map centered on Vancouver](~/images/display-map.png)

## Display markers on map

> After initialization of a maplibre Map, call the utility function [drawPoints](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#drawpoints) by passing in a `sourceName`, coordinate data, a maplibre-gl-js Map, styling options, and the map style. For a full list of styling options check the `drawPoints` [documentation](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#drawpoints)

Reviewer: ```suggestion
To display markers on a map, use the [drawPoints](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#drawpoints) function. `drawPoints` expects
- `sourceName` - specifies the layer on which the markers are rendered on. You can edit existing markers by passing the same `sourceName`
- coordinate data - the coordinate data of the markers to be displayed
- a maplibre-gl-js Map - the map object on which to render the markers
```

```javascript
async function initializeMap() {
    const map = await AmplifyMapLibreRequest.createMapLibreMap({
        container: "map", // An HTML Element or HTML element ID to render the map in https://maplibre.org/maplibre-gl-js-docs/api/map/
        center: [-123.1187, 49.2819],
        zoom: 11,
        region: "us-west-2"
    })
}

initializeMap();
```

![A map centered on Vancouver](~/images/display-map.png)

## Display markers on map

To display markers on a map, use the [drawPoints](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/API.md#drawpoints) function. `drawPoints` expects:

- `sourceName` - specifies the layer on which the markers are rendered on. You can edit existing markers by passing the same `sourceName`
- coordinate data - the coordinate data of the markers to be displayed
- a maplibre-gl-js Map - the map object on which to render the markers

<amplify-callout>

> The `drawPoints` method returns an object containing the string ids of the source and layers used to create the points on the map. These ids can be used to add any additional customizations through maplibre-gl-js [source](https://maplibre.org/maplibre-gl-js-docs/api/sources/), [paint](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#paint-property), and [layer](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/) options.

Reviewer: ```suggestion
The `drawPoints` method returns ids of the source and layers used to display the markers on the map. These ids can be used for further customization through maplibre-gl-js [source](https://maplibre.org/maplibre-gl-js-docs/api/sources/), [paint](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#paint-property), and [layer](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/) options.
```

</amplify-callout>

```javascript
map.on("load", function () {
    drawPoints("mySourceName", // Arbitrary source name
        [[-123.1187, 49.2819], [-122.849, 49.1913]], // Any coordinate or Feature data
        map,
        {
            showCluster: true,
            unclusteredOptions: {
                showMarkerPopup: true,
            },
            clusterOptions: {
                showCount: true,
            },
        }
    );
});

```
![A map with points](~/images/display-map-with-points.png)

## Display different map styles

> The `getAvailableMaps` api will fetch information for all maps that are available to be displayed.

Reviewer: ```suggestion
The `getAvailableMaps` API fetches information for all maps that are available to be displayed.
```

> In the context of Amazon Location Service maps, it will fetch the map name and the style of all maps that were generated using the Amplify CLI. This is useful is you would like to give your users a variety of maps styles to choose from.

Reviewer: ```suggestion
This is useful if you would like to give your users a variety of maps styles to choose from.
```

```javascript
geo.getAvailableMaps();
```

> This will return an array of maps that are available:

Reviewer: ```suggestion
The available maps are returned as an array with the following contents:
```

```javascript
//returns
[
  {
    mapName: 'myAmplifyGeoErsiStreetMap',
    style: 'VectorEsriStreets'
  },
  {
    mapName: 'myAmplifyGeoErsiTopographicMap',
    style: 'VectorEsriStreetsVectorEsriTopographic'
  },
]
```

> Then given you have an existing maplibre-gl Map you can use the `setStyle` and `resize` functions to update the displayed map

Reviewer: ```suggestion
You can resize and customize a map with the `resize` and `setStyle` functions:
```
